pandas/tests/groupby/test_apply.py

```diff
@@ -1065,7 +1065,9 @@ def test_apply_with_date_in_multiindex_does_not_convert_to_timestamp():
     )
 
     grp = df.groupby(["A", "B"])
-    result = grp.apply(lambda x: x.head(1))
+    with tm.assert_produces_warning(FutureWarning, check_stacklevel=False):
```

Comment:
we are exposing this to the user?

Reply:
any idea why this happens?

Reply:
when it tries to make a MultiIndex out of the A and B columns, it goes through Categorical, which passes convert_dates=True to maybe_infer_to_datetimelike,

Reply:
this should not be warnign at the top level

Reply:
we could finagle it so that we dont pass convert_dates=True to maybe_infer_to_datetimelike in this case, but that gets pretty complicated, especially compared to just warning the user that date objects are finicky

Reply:
Looks like a while back you were in [favor of](https://github.com/pandas-dev/pandas/issues/11058#issuecomment-139562271) telling people not to pass dates to the MultiIndex constructors.